File: day01/chronal_calibration.py
#!/usr/bin/env python3
from sys import argv

script, filename = argv
freq = 0
deltas = []
with open(filename) as f:
    for line in f:
        line = line.rstrip()
        freq += int(line)
        deltas.append(int(line))

print(f'resulting frequency = {freq}')

freq = 0
seen = set()
i = 0
while True:
    freq += deltas[i]
    if freq in seen:
        print(f'{freq} is first seen twice')
        break

    seen.add(freq)
    i = (i + 1) % len(deltas)
    

File: day01/chronal_calibration.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature 'say';

my $freq = 0;
my @deltas;
while (<>) {
    chomp;
    $freq += $_;
    push @deltas, $_;
}

say "resulting frequency = $freq";

$freq = 0;
my %seen;
my $i = 0;
while (1) {
    $freq += $deltas[$i];
    if (defined $seen{$freq}) {
	say "$freq is first seen twice";
	last;
    }
    $seen{$freq} = 1;
    $i = ($i + 1) % @deltas;
}
